src/ARCtrl/Conversion.fs

```diff
@@ -139,9 +149,64 @@ type BaseTypes =
         let dataType = LDFile.tryGetDisambiguatingDescriptionAsString(f, ?context = context) |> Option.map DataFile.fromString
         let format = LDFile.tryGetEncodingFormatAsString(f, ?context = context)
         let selectorFormat = LDFile.tryGetUsageInfoAsString(f, ?context = context)
-        let data = Data(id = f.Id, name = LDFile.getNameAsString(f, ?context = context), ?dataType = dataType, ?format = format, ?selectorFormat = selectorFormat)
+        let data = Data((*id = f.Id, *)name = LDFile.getNameAsString(f, ?context = context), ?dataType = dataType, ?format = format, ?selectorFormat = selectorFormat)
         data
 
+
+    static member composeFragmentDescriptor (dc : DataContext) =
+        if dc.Name.IsNone then failwith "RO-Crate parsing of DataContext failed: Cannot create a fragment descriptor without a name."
+        let id = LDPropertyValue.genIdFragmentDescriptor(dc.NameText)
+        let explicationName, explicationID =
+            dc.Explication
+            |> Option.map (fun e -> e.Name, e.TermAccessionAndOntobeeUrlIfShort |> Option.fromValueWithDefault "")
+            |> Option.defaultValue (None, None)
+        let unitName, unitID =
+            dc.Unit
+            |> Option.map (fun u -> u.Name, u.TermAccessionAndOntobeeUrlIfShort |> Option.fromValueWithDefault "")
+            |> Option.defaultValue (None, None)
+        let disambiguatingDescriptions = dc.Comments |> ResizeArray.map (fun c -> c.ToString()) |> Option.fromSeq
+        let dataFragment = BaseTypes.composeFile(dc)
+        let pattern = dc.ObjectType |> Option.map (BaseTypes.composeDefinedTerm)
+        dataFragment.SetProperty(LDFile.about, LDRef(id))
+        dataFragment.SetOptionalProperty(LDFile.pattern, pattern)
+        LDPropertyValue.createFragmentDescriptor(
+            dc.NameText,
+            ?value = explicationName,
+            ?valueReference = explicationID,
+            ?unitText = unitName,
+            ?unitCode = unitID,
+            ?measurementMethod = dc.GeneratedBy,
```

Comment:
This line produces empty properties in the ARC prototype. I think this should be catched somehow.